packages/odyssey-react-mui/src/labs/DateField.tsx

```diff
@@ -10,35 +10,60 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { ChangeEventHandler, ChangeEvent, memo, useCallback } from "react";
+import {
+  FocusEventHandler,
+  memo,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from "react";
 import { InputAdornment } from "@mui/material";
 import {
   DateField as MuiDateField,
   DateFieldProps as MuiDateFieldProps,
+  DateValidationError,
 } from "@mui/x-date-pickers";
 import { DateTime } from "luxon";
+import { useTranslation } from "react-i18next";
 
 import { Field, RenderFieldComponentProps } from "../Field";
 import { TextFieldProps } from "../TextField";
 
 export type DateFieldProps = {
-  defaultValue?: MuiDateFieldProps<DateTime>["defaultValue"];
-  onChange: ChangeEventHandler<HTMLInputElement>;
-  value?: MuiDateFieldProps<DateTime>["value"];
+  onChange?: (value: string) => void;
 } & Pick<
-  TextFieldProps,
-  | "endAdornment"
-  | "errorMessage"
-  | "hasInitialFocus"
-  | "hint"
-  | "id"
-  | "isDisabled"
-  | "isOptional"
-  | "isReadOnly"
-  | "label"
-  | "onBlur"
-  | "onFocus"
->;
+  MuiDateFieldProps<DateTime>,
+  "defaultValue" | "inputRef" | "minDate" | "maxDate" | "timezone" | "value"
+> &
+  Pick<
+    TextFieldProps,
+    | "endAdornment"
+    | "errorMessage"
+    | "hasInitialFocus"
+    | "hint"
+    | "id"
+    | "isDisabled"
+    | "isOptional"
+    | "isReadOnly"
+    | "label"
+    | "onBlur"
+    | "onFocus"
+  >;
+
+const useOdysseyDateError = () => {
+  const { t } = useTranslation();
+
+  return new Map<DateValidationError, string>([
+    ["invalidDate", t("picker.error.invalid")],
+    ["maxDate", t("picker.error.maxdate")],
+    ["minDate", t("picker.error.mindate")],
+  ]);
+};
+
+const formatDateTimeToUtcIsoDateString = (value: DateTime) =>
+  value.toUTC().toISO() || undefined;
```

Comment:
Why `|| undefined`? Is it `null` or `false`? It says in TS that it's a `DateTime`. The response has to be a string when you do `toISO()` right?